### PR TITLE
[CS2103T-T12-1] +Work

### PR DESCRIPTION
**Pull request by seahlynn**
@ArunBeCoding
@elsakoh
@gabrielseow
@ambhinav

### CODE REVIEW

File: docs/DeveloperGuide.adoc [hunk 1]
```diff
@@ -297,15 +299,22 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
 |Priority |As a ... |I want to ... |So that I can...
 |`* * *` |new user |see usage instructions |refer to instructions when I forget how to use the App
 
-|`* * *` |user |add a new person |
+|`* * *` |project leader |add a project member | asssign tasks to them
+
+|`* * *` |project leader |add tasks |
 
-|`* * *` |user |delete a person |remove entries that I no longer need
+|`* * *` |project leader |change task status | ger reminder of the progress of each task
```
Comment (SimNee) [inline docs/DeveloperGuide.adoc]: Small spelling error here

File: docs/DeveloperGuide.adoc [hunk 2]
```diff
@@ -315,33 +324,333 @@ _{More to be added}_
...
 +
-Use case ends.
+Use case ends
 
 *Extensions*
 
-[none]
-* 2a. The list is empty.
+2. The user does not specify name
+3. +work shows an error message
```
Comment (iskandarzulkarnaien) [inline docs/DeveloperGuide.adoc]: Numbering format is wrong. Should start from the step number where the extension occurs. Refer to [this](https://nus-cs2103-ay1920s1.github.io/website/schedule/week7/topics.html).

File: docs/DeveloperGuide.adoc [hunk 3]
```diff
@@ -315,33 +324,333 @@ _{More to be added}_
...
-[none]
-** 3a1. AddressBook shows an error message.
+Use case ends
+
+*Extensions*
+
+2. The list is empty
 +
-Use case resumes at step 2.
+Use case ends
```
Comment (iskandarzulkarnaien) [inline docs/DeveloperGuide.adoc]: I think it would be better to have the system feedback to the user that the list is empty instead of simply terminating

File: docs/DeveloperGuide.adoc [hunk 4]
```diff
@@ -297,15 +299,22 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
-|`* * *` |user |add a new person |
+|`* * *` |project leader |add a project member | asssign tasks to them
+
+|`* * *` |project leader |add tasks |
 
-|`* * *` |user |delete a person |remove entries that I no longer need
+|`* * *` |project leader |change task status | ger reminder of the progress of each task
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* * *` |project leader |sync team members schedules |Find a time slot where the maximum number of people, if not all, can attend
```
Comment (SimNee) [inline docs/DeveloperGuide.adoc]: The capitalisation should be consistent. "find" is capitalise but before that is not

File: docs/DeveloperGuide.adoc [hunk 5]
```diff
@@ -315,33 +324,333 @@ _{More to be added}_
...
 
-*MSS*
+*Main Success Scenario*
 
-1.  User requests to list persons
-2.  AddressBook shows a list of persons
-3.  User requests to delete a specific person in the list
-4.  AddressBook deletes the person
+1. User requests to add a team member
+2. User specifies the name of the team member
```
Comment (iskandarzulkarnaien) [inline docs/DeveloperGuide.adoc]: This seems to be that at step 1, the user types in the command, then +Work waits for the user to type in the name of the team member. May I suggest adding a step in between where `+Work notifies user to key in the name of team member`

File: docs/DeveloperGuide.adoc [hunk 6]
```diff
@@ -297,15 +299,22 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
-|`* * *` |user |delete a person |remove entries that I no longer need
+|`* * *` |project leader |change task status | ger reminder of the progress of each task
 
-|`* * *` |user |find a person by name |locate details of persons without having to go through the entire list
+|`* * *` |project leader |sync team members schedules |Find a time slot where the maximum number of people, if not all, can attend
+|`* *` |Project leader with many members and tasks|Find out which group members are working on a particular task |Distribute workload evenly
+|`* *` |Project leader with tasks that require inventory|assign tasks inventory or materials that are needed |Download an inventory report whenever required
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* *`|Project leader with frequent meetings|Create a new meeting with recommended time and desired location|View my next meeting in the home page
```
Comment (SimNee) [inline docs/DeveloperGuide.adoc]: Probably can split the function and benefit into 2 user stories since it does not seems to match.
As a project leader, I want to create meeting so that I can secure the meeting slot at my desired time and place.
As a project leader, I want to view my next meeting in the homepage so that I know my schedule at a glance.

File: docs/DeveloperGuide.adoc [hunk 7]
```diff
@@ -315,33 +324,333 @@ _{More to be added}_
...
+Use case ends
+
+[discrete]
+=== Use case: List all team members
 
-* 3a. The given index is invalid.
+*Main Success Scenario*
+
+1. User requests to list team members
+2. +work displays list of team members
```
Comment (iskandarzulkarnaien) [inline docs/DeveloperGuide.adoc]: Seems like your app name is inconsistent in its capitalization. Some places have `+work` and others have `+Work`. Best to make it consistent.

File: docs/DeveloperGuide.adoc (hunk 5 above)
Comment (iskandarzulkarnaien) [reply docs/DeveloperGuide.adoc]: This applies for your other use cases too

Comment (iskandarzulkarnaien) [reply docs/DeveloperGuide.adoc]: Also, before use case ends, it may be good to notify the user that their action was successful.

File: docs/DeveloperGuide.adoc [hunk 8]
```diff
@@ -297,15 +299,22 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
+|`* * *` |project leader |sync team members schedules |Find a time slot where the maximum number of people, if not all, can attend
+|`* *` |Project leader with many members and tasks|Find out which group members are working on a particular task |Distribute workload evenly
+|`* *` |Project leader with tasks that require inventory|assign tasks inventory or materials that are needed |Download an inventory report whenever required
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* *`|Project leader with frequent meetings|Create a new meeting with recommended time and desired location|View my next meeting in the home page
+
+|`*`|Project leader with a long-term project with ad-hoc members|Change members working on a task|Reassign tasks to incoming members
+
+|`*`|Project leader operating keeping track of purchases|Tag the inventory purchase to the member who bought it|Produce an accurate claims report at the end of the project
```
Comment (SimNee) [inline docs/DeveloperGuide.adoc]: "Project leader operating keeping track of purchases" I don't seem to understand this. Could you rephrase it in a better way?

File: docs/DeveloperGuide.adoc [hunk 9]
```diff
@@ -315,33 +324,333 @@ _{More to be added}_
...
+
++
+Use case ends
+
+[discrete]
+=== Use case:  Mark a task as ‘doing’
+
+*Main Success Scenario*
+
+1. User requests to mark a task as ‘doing
```
Comment (iskandarzulkarnaien) [inline docs/DeveloperGuide.adoc]: Missing close inverted comma

File: docs/DeveloperGuide.adoc [hunk 10]
```diff
@@ -273,19 +273,21 @@ Refer to the guide <<Testing#, here>>.
...
 * can type fast
+* prefers desktop apps over mobile apps
 * prefers typing over mouse input
 * is reasonably comfortable using CLI apps
 
-*Value proposition*: manage contacts faster than a typical mouse/GUI driven app
+*Value proposition*: manage tasks assigned to project mates, finding common time slots and keep track of inventory faster than GUI apps.
 
 [appendix]
 == User Stories
```
Comment (SimNee) [inline docs/DeveloperGuide.adoc:912]: Generally well structured and have all 3 parts present

File: docs/DeveloperGuide.adoc [hunk 11]
```diff
@@ -315,33 +324,333 @@ _{More to be added}_
...
+[discrete]
+=== Use case:  Remove a task from the dashboard
+
+*Main Success Scenario*
+
+1. User requests to remove a task from the dashboard
+2. User specifies the task ID
+3. +work updates the dashboard
++
+Use case ends
```
Comment (iskandarzulkarnaien) [inline docs/DeveloperGuide.adoc]: Seems to be duplicated by mistake

File: docs/DeveloperGuide.adoc [hunk 12]
```diff
@@ -297,15 +299,22 @@ Priorities: High (must have) - `* * \*`, Medium (nice to have) - `* \*`, Low (un
...
 
-|`* *` |user |hide <<private-contact-detail,private contact details>> by default |minimize chance of someone else seeing them by accident
+|`* *`|Project leader with frequent meetings|Create a new meeting with recommended time and desired location|View my next meeting in the home page
+
+|`*`|Project leader with a long-term project with ad-hoc members|Change members working on a task|Reassign tasks to incoming members
+
+|`*`|Project leader operating keeping track of purchases|Tag the inventory purchase to the member who bought it|Produce an accurate claims report at the end of the project
 
-|`*` |user with many persons in the address book |sort persons by name |locate a person easily
 |=======================================================================
```
Comment (SimNee) [inline docs/DeveloperGuide.adoc:945]: Did not really talk about the other type of users for this project. For example, project leader can assign tasks but did not talk about if the teammates can see the tasks

File: docs/DeveloperGuide.adoc [hunk 13]
```diff
@@ -315,33 +324,333 @@ _{More to be added}_
...
+Use case ends
+
+[discrete]
+=== Use case: Generate availability timings of team members
+
+*Main Success Scenario*
+
+1. User adds timetable of team mates
+2. User requests to generate availability timings of team members
+3. +work displays list of timings where the most number of team members are available
```
Comment (iskandarzulkarnaien) [inline docs/DeveloperGuide.adoc]: Same formatting issue here with extensions

File: docs/DeveloperGuide.adoc [hunk 14]
```diff
@@ -315,33 +324,333 @@ _{More to be added}_
...
+2. The list is empty.
+Use case ends
+
+3. The given index is invalid.
+4. +work shoes an error message
++
+Use case resumes at step 2
 
 [appendix]
 == Non Functional Requirements
```
Comment (SimNee) [inline docs/DeveloperGuide.adoc:1326]: 1. Are the NFRs are really Non-Functional?
- Yes because it is not related to any function. They are function requirements related to performance, user-friendliness. 

2. Is each NFR well-defined (i.e., possible to decide when it has been met)?

- NFR 1 is a well-defined NFR, but very generic
- NFR 2: it is ambiguous as to what ‘noticeable sluggishness’ means. How many seconds/milliseconds of lag would it be before we notice the sluggishness?
- NFR 3: it is a measurable NFR. Since the wording is ‘accomplish most of the tasks faster using commands’, are there any specific tasks where it would take longer to accomplish?
- Other feedback: the NFRs are quite generic, and not specific to the application. 

3. Is each NFR reasonably achievable?
- NFRs are reasonably achievable

4. Are more relevant NFRs have been left out?
- Left out NFRs related to security. A possible NFR would be: only team leader can change the allocation of tasks to team members.

**Review by iskandarzulkarnaien — COMMENTED** [review]
Good work in general! However there are formatting issues with use cases. Please refer to [this](https://nus-cs2103-ay1920s1.github.io/website/schedule/week7/topics.html)

File: docs/DeveloperGuide.adoc [hunk 15]
```diff
@@ -64,9 +63,9 @@ image::LogicClassDiagram.png[]
 [discrete]
```
Comment (weikiat97) [inline docs/DeveloperGuide.adoc:63]: Diagrams do not have updated class name (it is still named as AddressBookParser). I think the ProjectDashBoardParser should be connected to the XYZCommandParser in this case.

File: docs/DeveloperGuide.adoc [hunk 16]
```diff
@@ -98,9 +115,9 @@ image::LogicClassDiagram.png[]
 *API* :
```
Comment (weikiat97) [inline docs/DeveloperGuide.adoc:137]: The UI Component diagram is a bit difficult to understand. What are the functionalities of the UserView classes (Navigator / Controller / Main / Update)? Why is the UserViewMain connected to the Logic?

Comment (weikiat97) [reply docs/DeveloperGuide.adoc:137]: While the implementation may be unclear to us, the puml diagram is actually quite well done.

File: docs/DeveloperGuide.adoc [hunk 17]
```diff
@@ -109,33 +126,33 @@ Given below is the Sequence Diagram for interactions within the `Logic` componen
...
 
-NOTE: The lifeline for `DeleteCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
+NOTE: The lifeline for `DeleteTaskCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 
 [[Design-Model]]
 === Model component
 
 .Structure of the Model Component
-image::ModelClassDiagram.png[]
+image::ModelClassDiagramNew.png[]
```
Comment (weikiat97) [inline docs/DeveloperGuide.adoc:158]: Could consider putting the name/price/taskStatus under the attributes of the main classes.
Tag is an association, not a composition. Might want to change the notation for the tag.

File: docs/DeveloperGuide.adoc [hunk 18]
```diff
@@ -109,33 +126,33 @@ Given below is the Sequence Diagram for interactions within the `Logic` componen
...
 * stores the Address Book data.
-* exposes an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
+* exposes the unmodifiable `ObservableList<Member>`, `ObservableList<Task>`, `ObservableList<Inventory>` and `ObservableList<Mapping>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
 * does not depend on any of the other three components.
 
 [NOTE]
-As a more OOP model, we can store a `Tag` list in `Address Book`, which `Person` can reference. This would allow `Address Book` to only require one `Tag` object per unique `Tag`, instead of each `Person` needing their own `Tag` object. An example of how such a model may look like is given below. +
+As a more OOP model, we can store a `Tag` list in `Address Book`, which `Member` or 'Task' can reference. This would allow `Address Book` to only require one `Tag` object per unique `Tag`, instead of each `Member` or 'Task' needing their own `Tag` object. An example of how such a model may look like is given below. +
```
Comment (weikiat97) [inline docs/DeveloperGuide.adoc]: Don't forget to change the address books to your application! :)

File: docs/DeveloperGuide.adoc [hunk 19]
```diff
@@ -153,7 +170,184 @@ Classes used by multiple components are in the `seedu.addressbook.commons` packa
...
+
+These operations are exposed in the `Model` interface as `Model#generateTimings()`, `Model#getAvailableTimings()` and
+`Model#importCalendar()`.
+
+Given below is an example usage scenario and how the various commands work.
+
+Step 1. The user launches the application. The `ProjectCalendar` will be initialised based on the
+saved `ProjectCalendar`.
+
+Step 2. The user imports members' calendars by executing `impport-calendar`.
```
Comment (weikiat97) [inline docs/DeveloperGuide.adoc]: import (spelling error)
